src/core/include/iDynTree/Core/EigenHelpers.h

```diff
@@ -73,6 +73,43 @@ inline Eigen::Map<Eigen::VectorXd> toEigen(iDynTree::Span<double> vec)
 {
     return Eigen::Map<Eigen::VectorXd>(vec.data(),vec.size());
 }
+
+inline Eigen::Map<const Eigen::Matrix<double, Eigen::Dynamic, Eigen::Dynamic, Eigen::RowMajor>,
+                  0,
+                  Eigen::Stride<Eigen::Dynamic, Eigen::Dynamic>>
+toEigen(const MatrixView<const double>& mat)
+{
+    using MatrixRowMajor = Eigen::Matrix<double,Eigen::Dynamic,Eigen::Dynamic,Eigen::RowMajor>;
+
+    // This is a trick required to see a ColMajor matrix as a RowMajor matrix.
+    const int innerStride = (mat.storageOrder() == StorageOrder::ColMajor) ? mat.rows() : 1;
+    const int outerStride = (mat.storageOrder() == StorageOrder::ColMajor) ? 1 : mat.cols();
```

Comment:
I read [here](http://eigen.tuxfamily.org/dox-devel/TopicPitfalls.html) that it is not safe to use ternary operators with EigenTypes.

Reply:
Ah, my bad, Here mat is a MatrixView type!

Reply:
I think that it is not safe when an Eigen type is on the left side. In this case should be fine as only ``int`` are involved.

Reply:
Ah okay, understood.

Reply:
Here the operator is called on `int` so it shouldn't be a problem

Reply:
> I think that it is not safe when an Eigen type is on the left side. In this case should be fine as only int are involved.

Ops I missed the message